Entities/Activity.cpp

```diff
@@ -0,0 +1,1136 @@
+			// Check previous in line
+			m_ControlledActor[player] = g_MovableMan.GetPrevTeamActor(team, m_ControlledActor[player]);
+
+			// Looped around the whole actor chain, and couldn't find an available actor, so switch back to the original
+			if (m_ControlledActor[player] == prevActor) {
+				g_GUISound.UserErrorSound()->Play();
+				m_ControlledActor[player] = prevActor;
+				break;
+			}
+			// If we're gone around without even finding the one we started with (he might be dead), then try to switch to the brain, and if not that, then 0
+			if (m_ControlledActor[player] == start) {
+				if (m_Brain[player]) {
+					m_ControlledActor[player] = m_Brain[player];
+				} else if (g_MovableMan.IsActor(start)) {
+					m_ControlledActor[player] = start;
+				} else {
+					m_ControlledActor[player] = 0;
+				}
+				break;
+			}
+		}
+
+		// Now set the controller of the actor we switched to, and play switching sound effects
+		if (m_ControlledActor[player]) {
+			m_ControlledActor[player]->SetControllerMode(Controller::CIM_PLAYER, player);
+
+			// Play actor switching sound effects
+			(m_ControlledActor[player] == m_Brain[player]) ? g_GUISound.BrainSwitchSound()->Play() : g_GUISound.ActorSwitchSound()->Play();
+			// Only play air swoosh if actors are out of sight of each other
+			if (prevActor && Vector(prevActor->GetPos() - m_ControlledActor[player]->GetPos()).GetMagnitude() > g_FrameMan.GetResX() / 2) { g_GUISound.CameraTravelSound()->Play(); }			
+
+			ReassignSquadLeader(player, team);
+		}
+
+		// Follow the new guy normally
+		m_ViewState[player] = NORMAL;
+	}
+
+/////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+	void Activity::EnteredOrbit(Actor *orbitedCraft) {
+		if (!orbitedCraft) {
+			return;
+		}
+		int team = orbitedCraft->GetTeam();
+
+		// Find out cost multipliers for metagame, or we end up AI's earning money when their craft return if their nativeCostMultiplier is too low
+		float foreignCostMult = 1.0F;
+		float nativeCostMult = 1.0F;
+		if (g_MetaMan.GameInProgress()) {
+			for (int player = 0; player < MAXPLAYERCOUNT; player++) {
+				if (GetTeamOfPlayer(player) == orbitedCraft->GetTeam()) {
+					const MetaPlayer *metaPlayer = g_MetaMan.GetMetaPlayerOfInGamePlayer(player);
+					if (metaPlayer) {
+						foreignCostMult = metaPlayer->GetForeignCostMultiplier();
+						nativeCostMult = metaPlayer->GetNativeCostMultiplier();
+					}
+				}
+			}
+		}
+
+		// Did a brain just evacuate the Scene??
+		bool brainOnBoard = orbitedCraft->HasObjectInGroup("Brains");
+		// Total value of ship and cargo and crew
+		float totalValue = orbitedCraft->GetTotalValue(0, foreignCostMult, nativeCostMult);
+		// TODO ARGH WHAT IF TWO PLAYERS ON SAME TEAM ARE OF DIFFERENT TECHSS??!?!?!?$?!?!$?!$?
```

Comment:
I guess this todo is solved? I'm not sure what he means by the liquidation value, just the default value?

Seems to me the proper solution would be always use the lowest possible value for this sort of thing, so you'll never gain money, but that might be out of scope for this.